frontend/packages/core/src/timeago.tsx

```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import type { ReactTimeagoProps as TimeAgoProps } from "react-timeago";
+import ReactTimeago from "react-timeago";
+
+interface EventTimeProps extends Pick<TimeAgoProps, "date"> {
+  onClick?: () => void;
+  short?: boolean;
+  live?: boolean;
+}
+
+const unitFormatter = (unit: string): string => {
+  switch (unit) {
+    case "month":
+      // month -> mo
+      return unit.substring(0, 2);
+    default:
+      return unit.charAt(0);
+  }
+};
+
+const parseTimestamp = (timestamp?: number | Long | null): number => {
+  return parseInt(timestamp?.toString() || "0", 10);
+};
+
+const setMilliseconds = (timestamp?: number | Long | null): number => {
+  const ts = new Date(0);
+  return ts.setUTCMilliseconds(parseTimestamp(timestamp));
+};
+
+/**
+ * Will take a millsecond timestamp in and calculate the timeago for it
+ * @param date the millisecond timestamp
+ * @param short (default) will shorten the unit string (day -> d)
+ * @param live (default) will auto increment based on a given time
+ * @returns react component representing the timeago
+ */
+const TimeAgo = ({ short = true, onClick, ...props }: EventTimeProps) => (
```

Comment:
I'm wondering if it makes sense to have date be a `Date` type and then we don't have to rely on users passing in a number that may or may not be correct

Reply:
Actually it does seem to be a date type already from Timeago, so I've modified the comment.